### ISSUE TITLE
Suggest `(keep ...` instead of `(filter identity (map ...`

### ISSUE THREAD

**Issue by pmonks** [clj-kondo/clj-kondo#1440]
I have a bad habit of using idioms such as `(filter identity (map ...` when this is equivalent to `(keep ...`.  It would be great if kondo detected this former pattern and suggested the latter pattern when it sees it in code.

**Comment by pmonks** [comment]
I'm not sure why this issue would be closed, so as per the bot's instructions, here's a message to prevent that from happening.

**Comment by pmonks** [comment]
Upon further thought I don't think this issue is valid.  Here's a case that shows why:
```clojure
user=> (filter identity (map s/blank? [nil "" " " "foo" "bar"]))
(true true true)
user=> (keep s/blank? [nil "" " " "foo" "bar"])
(true true true false false)
```

This has to do with `keep`'s unusual behaviour when handling `false`; where "unusual" here means "not in keeping with most other core fns that refine sequences.

**Comment by stig** [comment]
Another example. I often see this: `(filter nil? (map #(...) xs))`, which I think is better expressed as `(keep #(...) xs)`. Would love that to be covered by this linter too.
